src/poketch_digital_clock/poketch_digital_clock_graphics.c

```diff
@@ -0,0 +1,251 @@
+#include "poketch_digital_clock/poketch_digital_clock_graphics.h"
+
+#include <nitro.h>
+#include <string.h>
+
+#include "overlay025/ov25_02254560.h"
+#include "overlay025/ov25_02255090.h"
+
+#include "bg_window.h"
+#include "graphics.h"
+#include "heap.h"
+#include "sys_task_manager.h"
+
+#define NUM_TASK_SLOTS 8
+
+#define POKETCH_CLOCK_DIGITS_TILEMAP_IDX 25
+#define POKETCH_NUM_CLOCK_DIGITS         10
+
+#define PRE_POKETCH_DISPLAY_NARC_TILES_IDX   23
+#define PRE_POKETCH_DISPLAY_NARC_TILEMAP_IDX 24
+
+#define TILE_SIZE_BYTES 2
+
+#define POKETCH_CLOCK_DIGIT_WIDTH     4
+#define POKETCH_CLOCK_DIGIT_HEIGHT    9
+#define POKETCH_CLOCK_DIGIT_GAP       1
+#define POKETCH_CLOCK_HOURS_DIGIT_X   3
+#define POKETCH_CLOCK_HOURS_DIGIT_Y   7
+#define POKETCH_CLOCK_MINUETS_DIGIT_X 15
+#define POKETCH_CLOCK_MINUETS_DIGIT_Y 7
+
+#define POKETCH_DIGIT_TILEMAP_WIDTH      (POKETCH_CLOCK_DIGIT_WIDTH * POKETCH_NUM_CLOCK_DIGITS)
+#define POKETCH_DIGIT_TILEMAP_HEIGHT     POKETCH_CLOCK_DIGIT_HEIGHT
+#define POKETCH_DIGIT_TILEMAP_SIZE_BYTES (POKETCH_DIGIT_TILEMAP_WIDTH * POKETCH_DIGIT_TILEMAP_HEIGHT)
+
+struct DisplayData_t {
+    const ClockData *clockData;
+    BgConfig *bgConfig;
+    u32 taskList[2 + NUM_TASK_SLOTS];
+    u16 digitsTilemap[POKETCH_DIGIT_TILEMAP_SIZE_BYTES];
+};
```

Comment:
```suggestion
typedef struct DisplayData {
    const ClockData *clockData;
    BgConfig *bgConfig;
    u32 taskList[2 + NUM_TASK_SLOTS];
    u16 digitsTilemap[POKETCH_DIGIT_TILEMAP_SIZE_BYTES];
} DisplayData;
```
and define this in the header file instead and remove the typedef there.

Reply:
I can move this but then these macros will need to be moved into the header even though they are not used outside of `*_graphics.c`

```
#define POKETCH_NUM_CLOCK_DIGITS 10

#define POKETCH_CLOCK_DIGIT_WIDTH  4
#define POKETCH_CLOCK_DIGIT_HEIGHT 9

#define POKETCH_DIGIT_TILEMAP_WIDTH      (POKETCH_CLOCK_DIGIT_WIDTH * POKETCH_NUM_CLOCK_DIGITS)
#define POKETCH_DIGIT_TILEMAP_HEIGHT     (POKETCH_CLOCK_DIGIT_HEIGHT)
#define POKETCH_DIGIT_TILEMAP_SIZE_BYTES (POKETCH_DIGIT_TILEMAP_WIDTH * POKETCH_DIGIT_TILEMAP_HEIGHT)
```

Reply:
That's generally fine; you can define a sister-header for the constants, if you'd like. If you do, store it as `include/constants/poketch/digital_watch.h`.